### PR TITLE
Decay habitat quality threats using euclidean distance transform

### LINKED ISSUE #1104: HQ exponential decay discrepancy
James mentioned that there is a discrepancy in how exponential decay is handled between source and the UG.

What the source code is doing: `exp(-dist / max_dist)`
What the UG says it should be doing: `exp(-(2.99 / max_dist) * dist)`

This came up from https://github.com/natcap/pygeoprocessing/issues/268#issuecomment-1299358823.

### PR DESCRIPTION
**Pull request by dcdenu4**
## Description
The Habitat Quality model has used convolutions as an implementation for decaying threat rasters over distance since InVEST 3.3.0. This approach strayed from the implementation described in the User's Guide and the User's Guide was never updated to reflect it. When thinking about updating the User's Guide to mention the convolution implementation it was not clear that the exponential decay via convolutions was providing the desired result. The degradation raster produced from the filtered threat rasters did not make intuitive sense either. Thus a decision was made to fall back to a euclidean distance transform implementation.

- Tests have been "eye" tested and updated.
- Model has been run using the Workbench UI and spot checked in GIS
- User's Guide has a PR here: https://github.com/natcap/invest.users-guide/pull/109

Fixes #1104 

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [x] Updated the user's guide (if needed)
- [x] Tested the affected models' UIs (if relevant)

### CODE REVIEW

File: HISTORY.rst
Comment (phargogh) [inline HISTORY.rst:47]:
```suggestion
      convolution implementation has been used, which reflected how
```

File: tests/test_habitat_quality.py
Comment (phargogh) [inline tests/test_habitat_quality.py:261]: Was this always writing to CWD before??

File: src/natcap/invest/habitat_quality.py
Comment (phargogh) [inline src/natcap/invest/habitat_quality.py:5]: Is `math` used?

Comment (phargogh) [reply src/natcap/invest/habitat_quality.py:5]: `flake8` is also suggesting that `osr` is imported but unused.

File: src/natcap/invest/habitat_quality.py
Comment (phargogh) [inline src/natcap/invest/habitat_quality.py:924]:
```suggestion
    The function will set pixels greater than ``max_dist`` to 0.

    Args:
        dist_raster_path (string): a filepath for the raster to decay.
            The raster is expected to be a euclidean distance transform with
            values measuring distance in pixels.
```

File: src/natcap/invest/habitat_quality.py
Comment (phargogh) [inline src/natcap/invest/habitat_quality.py:942]: Ah ok great, I see that these are already assumed to be square from the `align_and_resize_raster_stack` step.

File: src/natcap/invest/habitat_quality.py
Comment (phargogh) [inline src/natcap/invest/habitat_quality.py:1137]: Bonus newline?

File: tests/test_habitat_quality.py
Comment (phargogh) [inline tests/test_habitat_quality.py:1053]: Was this left over from debugging, or should we still be asserting this?

**Review by phargogh — CHANGES_REQUESTED** [review]
Looks sensible to me, @dcdenu4 !  I had a variety of trivial edits to suggest, and then it looks like the HQ tests are failing on something that looks like a missing raster.

Would you like to wait until https://github.com/natcap/invest.users-guide/pull/109 is merged and the UG revision here updated?

File: tests/test_habitat_quality.py
Comment (dcdenu4) [reply tests/test_habitat_quality.py:261]: Haha, yes!!!

File: src/natcap/invest/habitat_quality.py
Comment (dcdenu4) [reply src/natcap/invest/habitat_quality.py:5]: Thanks, I need to incorporate flake8 back into my workflow

File: src/natcap/invest/habitat_quality.py
Comment (dcdenu4) [reply src/natcap/invest/habitat_quality.py]: Fixed https://github.com/natcap/invest/pull/1159/commits/30ba06614bd613c6870a36a4b8b4a0562cf2454b

File: src/natcap/invest/habitat_quality.py
Comment (dcdenu4) [reply src/natcap/invest/habitat_quality.py]: Fixed https://github.com/natcap/invest/pull/1159/commits/30ba06614bd613c6870a36a4b8b4a0562cf2454b

File: tests/test_habitat_quality.py
Comment (dcdenu4) [reply tests/test_habitat_quality.py]: Huh, it must have been from debugging the new test results. It's a duplicate to an assertion right below it. I removed it. https://github.com/natcap/invest/pull/1159/commits/85a6d23583e268313d5838cea43b069a18882d60

**Comment by dcdenu4** [comment]
Hey @phargogh, I fixed the breaking test in https://github.com/natcap/invest/pull/1159/commits/8d177219feb59926e176589ddbccf8d9b8f22c60. I think I addressed all your other comments as well.

I do think it'd be nice to wait for the UG PR to merge to be able to link this to that revision. I'm just waiting for Stacie to give me a thumbs up on a new figure.

**Comment by dcdenu4** [comment]
@phargogh, I've updated the Makefile with the UG revision corresponding to the merged UG changes!

**Comment by dcdenu4** [comment]
Just noting that we're merging this into `main` where as the UG revision is from UG release branch 3.13.0 where GLOBIO has been removed. GLOBIO is not removed in InVEST `main` yet though, it's sitting in the InVEST 3.13.0 branch. In case that breaks anything...

**Comment by phargogh** [comment]
@dcdenu4 There's still one action failure, when building/validating sample data.  Other than that, I think this is good to go!

**Comment by dcdenu4** [comment]
@phargogh, that failure is from the GLOBIO mismatch between InVEST `main` (GLOBIO exists still) and UG `3.13.0` (GLOBIO is removed). I could switch the base in this PR to merge into InVEST `3.13.0` instead of `main`... I'll do that.

**Comment by dcdenu4** [comment]
Wait for PR #1206 before resolving conflicts and merging.

**Comment by dcdenu4** [comment]
Hey @phargogh , I _THINK_ that once the rest of the tests pass this should be good to go. 🤞

**Pull request merged**
